### PR TITLE
[RNMobile] Add Playback Panel to VideoPress Block Settings

### PR DESCRIPTION
**Pull request by SiobhyB**
Part of an ongoing project to add the VideoPress block to the native apps.

## Proposed changes:

* A native version of the `PlaybackPanel` component has been introduced and added to the VideoPress block's settings. The main code changes in the component's file can be found in https://github.com/Automattic/jetpack/pull/29457/commits/9f149cf9766ac0f2936d8d0d3615d4d137083917.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p9ugOq-3mV-p2 

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

* Go through the steps to add the VideoPress block on the web, optionally toggle the playback settings.
* Open up the app and navigate to the block that was added via the web.
* Tap the settings toggle to verify and then tap `Playback Settings` to open the new playback settings panel.
* If you'd changed the video's playback settings on the web, verify that these are accurately reflected on native.
* Experiment with toggling the settings and ensuring they work as expected.

### Screenshots

| 1st level  | 2nd level |
| ------------- | ------------- |
|  |  |

### Screen recording

The following screen recording shows the extra helper text/warning that displays when the autoplay setting is enabled:

https://user-images.githubusercontent.com/2998162/225115829-3414c06b-facd-4586-870b-f8fa1897cd64.mov

### CODE REVIEW

**Comment by SiobhyB** [comment]
👋 @osullivanchris, I wondered if I could get a design review from you before submitting for code review. I've added all of the same helper text that's displayed on the web, as we discussed, and wanted to check whether you think this looks okay or if it perhaps occupies too much screen time.

I also copied the way the web hides some of the autoplay helper text until it's toggled on (as shown in the PR's screen recording). 

Looking forward to hearing your thoughts on the way this looks and/or any potential changes!

**Review by osullivanchris — DISMISSED** [review]
> 👋 @osullivanchris, I wondered if I could get a design review from you before submitting for code review. 

Yes, please! I'm always keen to review builds and discuss details any time, the earlier the better.


> I've added all of the same helper text that's displayed on the web, as we discussed, and wanted to check whether you think this looks okay or if it perhaps occupies too much screen time.

Apologies, I forgot to go back and mock that up. But this looks as I'd expect. The extra information is not bothering me. I think its probably a good thing that it has its own deeper view now, where its focused on playback controls. It may have felt messy mixed in with other controls, with all this information.

> I also copied the way the web hides some of the autoplay helper text until it's toggled on (as shown in the PR's screen recording).

I didn't expect this. And was slightly unsure, thinking it could just display the full text all the time. But its kinda cool that it draws my attention to that warning, when I turn it on. So I'm good with what you've done!


Let me know if any other input required. By the way I'm reviewing this just based on screenshots and video as I'm not sure if there's a build available. But happy to install and test it as well.

**Comment by SiobhyB** [comment]
Thank you @osullivanchris 🙌 

> Let me know if any other input required. By the way I'm reviewing this just based on screenshots and video as I'm not sure if there's a build available. But happy to install and test it as well.

We don't have builds available at the moment, as there's a bit of a process for creating them for Mobile Gutenberg. I can create builds once all the block settings are merged, though, and it'd be great to get your input from them 🙇‍♀️

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx
Comment (jhnstn) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx]: Curious if you considered `useCallback` for `goBack()` and `openSubSheet()` . They are small but they do end up being recreated on each render.

**Review by jhnstn — DISMISSED** [review]
Test plan worked for me. I just left one comment but it's a non-blocker. Nice work!

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx
Comment (SiobhyB) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx]: Good catch, added in https://github.com/Automattic/jetpack/pull/29457/commits/28be169095e54b6905063d67a3378c5f277f8c1e. Let me know what you think :)

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx]: I'm wondering if we should add another new-line to separate the note from the help description, WDYT?
cc @osullivanchris 

Current|Proposal
--|--
|

**Review by fluiddot — DISMISSED** [review]
LGTM 🎊 !

I followed the Testing instructions on both platforms and worked as expected, great job 🏅  ! I only added a minor comment that shouldn't block the PR.

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx
Comment (osullivanchris) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx]: @fluiddot good idea, that looks slightly better to me.

Comment (SiobhyB) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx]: Sounds great to me! I've gone ahead to implement this in https://github.com/Automattic/jetpack/pull/29457/commits/5748198cfc64fe629b2e8c4b74a3f1b0b391fa95:

| Before  | After |
| ------------- | ------------- |
|  |  |

@fluiddot, if this looks good to you, would you be able you be able to approve the PR again for merge? Thanks!

Comment (fluiddot) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/playback-panel/index.native.tsx]: Great, thanks for updating this 🙇  .

> @fluiddot, if this looks good to you, would you be able you be able to approve the PR again for merge? Thanks!

Sure, I'll approve it now.

**Review by fluiddot — DISMISSED** [review]
LGTM 🎊 !

File: projects/plugins/jetpack/changelog/rnmobile-videopress-block-add-playback-settings
Comment (retrofox) [inline projects/plugins/jetpack/changelog/rnmobile-videopress-block-add-playback-settings:3]: I'm not the ideal person to say this, but still, the comment should reflect the changes introduced here :-)

Comment (dhasilva) [reply projects/plugins/jetpack/changelog/rnmobile-videopress-block-add-playback-settings:3]: This is the jetpack changelog. The comments are in the VideoPress package changelog.

Comment (SiobhyB) [reply projects/plugins/jetpack/changelog/rnmobile-videopress-block-add-playback-settings:3]: @retrofox, I didn't add this one, it was automatically generated. :/

Comment (fluiddot) [reply projects/plugins/jetpack/changelog/rnmobile-videopress-block-add-playback-settings:3]: @SiobhyB Not sure if the last commits were needed after merging `trunk` into this branch, I'm wondering why we need to [upgrade the VideoPress package](https://github.com/Automattic/jetpack/pull/29457/commits/78e52d7b2d7fae1a8a5d9c45ade8488fb2d84bf4) version and [upgrade the version in the Forms package](https://github.com/Automattic/jetpack/pull/29457/commits/e27a3e7728c69a87766fe0294ee734300e9e7b99) 🤔.

Comment (SiobhyB) [reply projects/plugins/jetpack/changelog/rnmobile-videopress-block-add-playback-settings:3]: @fluiddot, I asked about this more at p1679412895593819-slack-C03TA48NSUX and p1679413895219569-slack-C03TA48NSUX, which might be a good place to ask any extra questions from the Jetpack team.

Comment (dhasilva) [reply projects/plugins/jetpack/changelog/rnmobile-videopress-block-add-playback-settings:3]: I'm not sure why it is necessary to update the versions here, but sometimes when opening a PR right after some project was updated we need to update its version for the next alpha.
There was a `minor` VideoPress package update very recently, so maybe that is the reason? :man_shrugging: 

In any case, the `Linting / Changelogger validity (pull_request)` test fails whenever it detects that some version change is needed for any reason.

**Review by dhasilva — APPROVED** [review]
Code LGTM! :rocket:

**Pull request merged**
